01_classify_stratified/nbconverted/plot_cancer_type_results.py

```diff
@@ -228,3 +228,58 @@ def label_points(x, y, gene, sig, ax):
                            plt.gca())
 adjust_text(text_labels, ax=plt.gca())
 
+
+# ## Confusion matrix
+
+# In[10]:
+
+
+import os
+
+import mpmp.utilities.data_utilities as du
+
+preds_dir = os.path.join(cfg.repo_root, 'results_preds', 'cancer_type')
+sample_info_df = du.load_sample_info()
+
+preds_expression_df = au.load_preds_to_matrix(preds_dir, sample_info_df,
+                                              training_data='expression')
+print(preds_expression_df.shape)
+preds_expression_df.iloc[:5, :5]
+
+
+# In[11]:
+
+
+sns.set({'figure.figsize': (15, 10)})
+ax = sns.heatmap(preds_expression_df, cbar_kws={'label': 'Predicted probability of positive label, averaged over samples'})
+# this is needed to increase colorbar label font size
+# https://stackoverflow.com/a/48587137
+ax.figure.axes[-1].yaxis.label.set_size(14)
+plt.xlabel('True cancer type label', size=14)
+plt.ylabel('Positive label used to train classifier', size=14)
+plt.title('Cancer type confusion matrix, gene expression data', size=14, pad=14)
+plt.tight_layout()
+
+
+# In[12]:
+
+
+preds_methylation_df = au.load_preds_to_matrix(preds_dir, sample_info_df,
+                                               training_data='methylation')
+print(preds_methylation_df.shape)
+preds_methylation_df.iloc[:5, :5]
+
+
+# In[13]:
+
+
+sns.set({'figure.figsize': (15, 10)})
```

Comment:
Definitely a lot more noise in the methylation data. 

* Looks like methylation data is not as predictive within the same cancer type compared to using expression data. However there does appear to be more cross information gained. Looks like you're working on some clustering analysis, I wonder if those off diagonal entries are from the same tissue type -- is methylation data very tissue specific?

Reply:
Interesting...so maybe gene expression data is most predictive when you're within the same context, but methylation can be helpful if you are trying to transfer information to a new context!

Reply:
> * Looks like methylation data is not as predictive within the same cancer type compared to using expression data. However there does appear to be more cross information gained. Looks like you're working on some clustering analysis, I wonder if those off diagonal entries are from the same tissue type -- is methylation data very tissue specific?

Yeah, I'll have to look into it! I think it does tend to be tissue-specific, but I'm not sure (and I'm not sure if I'd expect it to be more or less tissue-specific than expression).

> Interesting...so maybe gene expression data is most predictive when you're within the same context, but methylation can be helpful if you are trying to transfer information to a new context!

We were thinking along these lines too...stay tuned!